src/controllers/authenticationController.ts

```diff
@@ -464,22 +464,92 @@ export default class AuthenticationController {
-
-        if (E.isLeft(errorOrResponse)) {
-          const error = errorOrResponse.left;
-          return ResponseErrorInternal(error.message);
-        }
-
-        const response = errorOrResponse.right;
-
-        if (!response) {
-          return ResponseErrorInternal("Error destroying the user session");
-        }
-
-        return ResponseSuccessJson({ message: "ok" });
-      })
+      withCatchAsInternalError(async () =>
+        pipe(
+          this.lollipopParams.isLollipopEnabled,
+          O.fromPredicate(identity),
+          // if lollipop is enabled we delete the reference and send the revoke pub key message
+          O.map(_ =>
+            pipe(
+              // retrieve the assertionRef for the user
+              TE.tryCatch(
+                async () =>
+                  await this.sessionStorage.getLollipopAssertionRefForUser(
+                    user
+                  ),
+                E.toError
+              ),
+              TE.chainEitherK(identity),
+              TE.chainW(maybeAssertionRef =>
+                pipe(
+                  // delete the assertionRef for the user
+                  TE.tryCatch(
+                    async () =>
+                      await this.sessionStorage.delLollipopAssertionRefForUser(
+                        user.fiscal_code
+                      ),
+                    E.toError
+                  ),
+                  TE.chainEitherK(identity),
+                  TE.chainW(_ =>
+                    pipe(
+                      maybeAssertionRef,
+                      O.map(assertionRef =>
+                        pipe(
+                          // send the revoke pubkey message with the assertionRef for the user
+                          TE.tryCatch(
+                            async () =>
+                              await this.lollipopParams.lollipopService.revokePreviousAssertionRef(
+                                assertionRef
+                              ),
+                            E.toError
+                          ),
+                          TE.chain(
+                            TE.fromPredicate(
+                              response => response.errorCode === undefined,
+                              response => new Error(response.errorCode)
+                            )
+                          ),
+                          TE.map(__ => true)
+                        )
+                      ),
+                      // ignore if there's no assertionRef on redis
+                      O.getOrElseW(() => TE.of(true))
+                    )
+                  )
+                )
+              )
+            )
+          ),
+          // if lollipop is not enabled we go on
+          O.getOrElse(() => TE.of(true)),
+          TE.chain(_ =>
+            pipe(
+              // delete the session for the user
+              TE.tryCatch(
+                async () => await this.sessionStorage.del(user),
```

Comment:
```suggestion
                () => this.sessionStorage.del(user),
```

Reply:
did in https://github.com/pagopa/io-backend/pull/985/commits/72ad564ab513c283fe3eedd1bfc2b09dd75f6390